File: app/src/main/java/ru/marslab/marsnotes/domain/Publisher.java
package ru.marslab.marsnotes.domain;

import java.util.ArrayList;
import java.util.List;

public class Publisher {

    private final List<Observer> observers = new ArrayList<>();

    public void subscribe(Observer observer) {
        observers.add(observer);
    }

    public void unSubscribe(Observer observer) {
        observers.remove(observer);
    }

    public void notify(int noteId) {
        for (Observer observer : observers) {
            observer.updateNoteById(noteId);
        }
    }

}

File: app/src/main/java/ru/marslab/marsnotes/domain/Publisher.kt
package ru.marslab.marsnotes.domain

import ru.marslab.marsnotes.domain.model.Note

class Publisher {
    private val observers = mutableListOf<Observer>()
    fun subscribe(observer: Observer) {
        observers.add(observer)
    }

    fun unSubscribe(observer: Observer) {
        observers.remove(observer)
    }

    fun notify(note: Note) {
        observers.forEach { it.updateNote(note) }
    }
}
